firestore/integration_test.go

```diff
@@ -1674,12 +1674,26 @@ func TestIntegration_ColGroupRefPartitionsLarge(t *testing.T) {
 	// Verify that we retrieve 383 documents across all partitions. (128*2 + 127)
 	totalCount := 0
 	for _, query := range partitions {
-
 		allDocs, err := query.Documents(ctx).GetAll()
 		if err != nil {
 			t.Fatalf("GetAll(): received unexpected error: %v", err)
 		}
 		totalCount += len(allDocs)
+
+		// Check that the same results are returned even if we use the proto converted query
+		pbStructuredQuery, _ := query.ToProto()
+		q := *iClient.Query()
+		q, err = q.FromProto(pbStructuredQuery)
+		if err != nil {
+			t.Fatalf("FromProto error: %v", err)
+		}
+		protoReturnedDocs, err := q.Documents(ctx).GetAll()
+		if err != nil {
+			t.Fatalf("GetAll error: %v", err)
+		}
+		if len(allDocs) != len(protoReturnedDocs) {
+			t.Fatalf("Expected document count to be the same on both query runs: %v", err)
+		}
```

Comment:
FYI Since this is a partition query, this does not verify a lot of the query internals (e.g. filters or more complex cursors, orderBys).

Reply:
Yep. Most of what I am using to test the change is our existing toproto test which I modified to do a roundtrip verification